src/statsbase.jl

```diff
@@ -0,0 +1,186 @@
+        rem_dim = first(setdiff((1, 2), numerical_dim))
+        rem_name = dimnames(A, rem_dim)
+        rem_key = axiskeys(A, rem_dim)
+        return KeyedArray(
+            NamedDimsArray(data, (rem_name, rem_name)),
+            (copy(rem_key), copy(rem_key))
+        )
+    else
+        data = cor(parent(A), wv, dims; kwargs...)
+        # Use same remaining axis for both dimensions of data
+        rem_dim = first(setdiff((1, 2), dims))
+        new_keys = Tuple(copy(axiskeys(A, rem_dim)) for i in 1:2)
+        return KeyedArray(data, new_keys)
+    end
+end
+
+function Statistics.cor(A::KeyedArray, wv::KeyedVector; kwargs...)
+    return cor(A, hasnames(wv) ? parent(parent(wv)) : parent(wv); kwargs...)
+end
+
+# Similar to cov and cor, but we aren't extending Statistics
+function StatsBase.scattermat(A::KeyedMatrix, wv::Vararg{<:AbstractWeights}; dims=1, kwargs...)
+    # A little awkward, but the weighted `cov` method from statsbase only works
+    # on dense matrices, so we need to unwrap this twice and completely rebuild the
+    # array
+    if AxisKeys.hasnames(A)
+        numerical_dim = NamedDims.dim(dimnames(A), dims)
+        data = scattermat(parent(parent(A)), wv...; dims=numerical_dim, kwargs...)
+        rem_dim = first(setdiff((1, 2), numerical_dim))
+        rem_name = dimnames(A, rem_dim)
+        rem_key = axiskeys(A, rem_dim)
+        return KeyedArray(
+            NamedDimsArray(data, (rem_name, rem_name)),
+            (copy(rem_key), copy(rem_key))
+        )
+    else
+        data = scattermat(parent(A), wv...; dims=dims, kwargs...)
+        # Use same remaining axis for both dimensions of data
+        rem_dim = first(setdiff((1, 2), dims))
+        new_keys = Tuple(copy(axiskeys(A, rem_dim)) for i in 1:2)
+        return KeyedArray(data, new_keys)
+    end
+end
+
+function StatsBase.scattermat(A::KeyedArray, wv::KeyedVector; kwargs...)
+    return scattermat(A, hasnames(wv) ? parent(parent(wv)) : parent(wv); kwargs...)
+end
+
+function StatsBase.mean_and_std(A::KeyedMatrix, wv::Vararg{<:AbstractWeights}; dims=:, corrected::Bool=true, kwargs...)
+    return (
+        mean(A, wv...; dims=dims, kwargs...),
+        std(A, wv...; dims=dims, corrected=corrected, kwargs...)
+    )
+end
+function StatsBase.mean_and_std(A::KeyedArray, wv::KeyedVector; kwargs...)
+    return mean_and_std(A, hasnames(wv) ? parent(parent(wv)) : parent(wv); kwargs...)
+end
+
+function StatsBase.mean_and_var(A::KeyedMatrix, wv::Vararg{<:AbstractWeights}; dims=:, corrected::Bool=true, kwargs...)
+    return (
+        mean(A, wv...; dims=dims, kwargs...),
+        var(A, wv...; dims=dims, corrected=corrected, kwargs...)
+    )
+end
+function StatsBase.mean_and_var(A::KeyedArray, wv::KeyedVector; kwargs...)
+    return mean_and_var(A, hasnames(wv) ? parent(parent(wv)) : parent(wv); kwargs...)
+end
+
+function StatsBase.mean_and_cov(A::KeyedMatrix, wv::Vararg{<:AbstractWeights}; dims=:, corrected::Bool=true, kwargs...)
```

Comment:
Can these `mean_and_x` funtions also be done in an `@eval` loop?

Reply:
Yeah, I think if I just build up the full func name in the loop then we should be good.